lang/cs/Org.Apache.REEF.Common/Runtime/Evaluator/EvaluatorRuntime.cs

```diff
@@ -241,6 +250,11 @@ private void OnException(Exception e)
             }
         }
 
+        private void UnhandledExceptionHandler(object sender, UnhandledExceptionEventArgs e)
```

Comment:
I suggest to make this an injectable class. That way, Tasks that want to spawn threads can use the same exception handler for them.